homeassistant/components/climate/ecobee.py

```diff
@@ -246,25 +264,38 @@ def turn_away_mode_off(self):
 
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
-        if kwargs.get(ATTR_TARGET_TEMP_LOW) is not None and \
-           kwargs.get(ATTR_TARGET_TEMP_HIGH) is not None:
-            high_temp = int(kwargs.get(ATTR_TARGET_TEMP_LOW))
-            low_temp = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))
+        low_temp = None
+        high_temp = None
+        temp = None
+
+        if kwargs.get(ATTR_TARGET_TEMP_LOW) is not None:
+            low_temp = int(kwargs.get(ATTR_TARGET_TEMP_LOW))
+
+        if kwargs.get(ATTR_TARGET_TEMP_HIGH) is not None:
+            high_temp = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))
+
+        if kwargs.get("temperature") is not None:
+            temp = int(kwargs.get("temperature"))
+
+        if self.current_operation == STATE_HEAT:
+            low_temp = temp
+            high_temp = temp + 20
+        elif self.current_operation == STATE_COOL:
+            low_temp = temp - 20
+            high_temp = temp
```

Comment:
I didn't add the hold_temp condition. I just copied that from the existing code. I'm happy to discuss, but I think you know better how the device actually works. I don't own an ecobee, I've just had a brief look at the ecobee API to try and understand existing code here in home assistant.